tests/ci/benchmark_framework/compare_results.py

```diff
@@ -15,6 +58,12 @@ def read_data(file):
         df = pd.read_csv(file, skiprows=1, index_col=0)
     return df
 
+def significant_regressions(compared_df):
+    if compared_df.empty:
+        return False
+    descriptions = compared_df['description.1']
+    evaluation = np.isin(descriptions, ignored_fields)
+    return not np.all(evaluation)
```

Comment:
Does `description.1` mean the 1st column of the csv file generated by speed tool? Can `descriptions = compared_df['description.1']` return non `TrustToken` fields?

Reply:
`description.1` is the description from the first file (aka the prod test run). All the names have either a `1` or a `2` appended depending on whether it's from the prod or pr run. I have to use 1 of them but they will always be equal so I picked the first one without loss of generality.

Reply:
Can `descriptions = compared_df['description.1']` return non `TrustToken` values?

Reply:
Yes, those are just the ones that I wanted to skip for now.

Reply:
Here's the list of values for the aws-lc benchmarks:
```
AES-128 encrypt setup, AES-128 encrypt, AES-128 decrypt setup, AES-128 decrypt, AES-192 encrypt setup, AES-192 encrypt, AES-192 decrypt setup, AES-192 decrypt, AES-256 encrypt setup, AES-256 encrypt, AES-256 decrypt setup, AES-256 decrypt, AES-256-XTS (16 bytes) init and encrypt, AES-256-XTS (256 bytes) init and encrypt, AES-256-XTS (1350 bytes) init and encrypt, AES-256-XTS (8192 bytes) init and encrypt, AES-256-XTS (16384 bytes) init and encrypt, AES-256-XTS (16 bytes) init and decrypt, AES-256-XTS (256 bytes) init and decrypt, AES-256-XTS (1350 bytes) init and decrypt, AES-256-XTS (8192 bytes) init and decrypt, AES-256-XTS (16384 bytes) init and decrypt, MD4 (16 bytes), MD4 (256 bytes), MD4 (1350 bytes), MD4 (8192 bytes), MD4 (16384 bytes), MD5 (16 bytes), MD5 (256 bytes), MD5 (1350 bytes), MD5 (8192 bytes), MD5 (16384 bytes), SHA-1 (16 bytes), SHA-1 (256 bytes), SHA-1 (1350 bytes), SHA-1 (8192 bytes), SHA-1 (16384 bytes), sha-224 (16 bytes), sha-224 (256 bytes), sha-224 (1350 bytes), sha-224 (8192 bytes), sha-224 (16384 bytes), SHA-256 (16 bytes), SHA-256 (256 bytes), SHA-256 (1350 bytes), SHA-256 (8192 bytes), SHA-256 (16384 bytes), SHA-384 (16 bytes), SHA-384 (256 bytes), SHA-384 (1350 bytes), SHA-384 (8192 bytes), SHA-384 (16384 bytes), SHA-512 (16 bytes), SHA-512 (256 bytes), SHA-512 (1350 bytes), SHA-512 (8192 bytes), SHA-512 (16384 bytes), HMAC-MD5 (16 bytes), HMAC-MD5 (256 bytes), HMAC-MD5 (1350 bytes), HMAC-MD5 (8192 bytes), HMAC-MD5 (16384 bytes), HMAC-SHA1 (16 bytes), HMAC-SHA1 (256 bytes), HMAC-SHA1 (1350 bytes), HMAC-SHA1 (8192 bytes), HMAC-SHA1 (16384 bytes), HMAC-SHA256 (16 bytes), HMAC-SHA256 (256 bytes), HMAC-SHA256 (1350 bytes), HMAC-SHA256 (8192 bytes), HMAC-SHA256 (16384 bytes), HMAC-SHA384 (16 bytes), HMAC-SHA384 (256 bytes), HMAC-SHA384 (1350 bytes), HMAC-SHA384 (8192 bytes), HMAC-SHA384 (16384 bytes), HMAC-SHA512 (16 bytes), HMAC-SHA512 (256 bytes), HMAC-SHA512 (1350 bytes), HMAC-SHA512 (8192 bytes), HMAC-SHA512 (16384 bytes), HMAC-MD5-OneShot (16 bytes), HMAC-MD5-OneShot (256 bytes), HMAC-MD5-OneShot (1350 bytes), HMAC-MD5-OneShot (8192 bytes), HMAC-MD5-OneShot (16384 bytes), HMAC-SHA1-OneShot (16 bytes), HMAC-SHA1-OneShot (256 bytes), HMAC-SHA1-OneShot (1350 bytes), HMAC-SHA1-OneShot (8192 bytes), HMAC-SHA1-OneShot (16384 bytes), HMAC-SHA256-OneShot (16 bytes), HMAC-SHA256-OneShot (256 bytes), HMAC-SHA256-OneShot (1350 bytes), HMAC-SHA256-OneShot (8192 bytes), HMAC-SHA256-OneShot (16384 bytes), HMAC-SHA384-OneShot (16 bytes), HMAC-SHA384-OneShot (256 bytes), HMAC-SHA384-OneShot (1350 bytes), HMAC-SHA384-OneShot (8192 bytes), HMAC-SHA384-OneShot (16384 bytes), HMAC-SHA512-OneShot (16 bytes), HMAC-SHA512-OneShot (256 bytes), HMAC-SHA512-OneShot (1350 bytes), HMAC-SHA512-OneShot (8192 bytes), HMAC-SHA512-OneShot (16384 bytes), RNG (16 bytes), RNG (256 bytes), RNG (1350 bytes), RNG (8192 bytes), RNG (16384 bytes), ECDH P-224, ECDH P-256, ECDH P-384, ECDH P-521, ECDH secp256k1, ECDSA P-224 signing, ECDSA P-224 verify, ECDSA P-256 signing, ECDSA P-256 verify, ECDSA P-384 signing, ECDSA P-384 verify, ECDSA P-521 signing, ECDSA P-521 verify, ECDSA secp256k1 signing, ECDSA secp256k1 verify, scrypt (N = 1024, r = 8, p = 16), scrypt (N = 16384, r = 8, p = 1), RSA 2048 signing, RSA 2048 verify (same key), RSA 2048 verify (fresh key), RSA 2048 private key parse, RSA 4096 signing, RSA 4096 verify (same key), RSA 4096 verify (fresh key), RSA 4096 private key parse, AES-128-GCM (16 bytes) seal, AES-128-GCM (256 bytes) seal, AES-128-GCM (1350 bytes) seal, AES-128-GCM (8192 bytes) seal, AES-128-GCM (16384 bytes) seal, AES-256-GCM (16 bytes) seal, AES-256-GCM (256 bytes) seal, AES-256-GCM (1350 bytes) seal, AES-256-GCM (8192 bytes) seal, AES-256-GCM (16384 bytes) seal, ChaCha20-Poly1305 (16 bytes) seal, ChaCha20-Poly1305 (256 bytes) seal, ChaCha20-Poly1305 (1350 bytes) seal, ChaCha20-Poly1305 (8192 bytes) seal, ChaCha20-Poly1305 (16384 bytes) seal, DES-EDE3-CBC-SHA1 (16 bytes) seal, DES-EDE3-CBC-SHA1 (256 bytes) seal, DES-EDE3-CBC-SHA1 (1350 bytes) seal, DES-EDE3-CBC-SHA1 (8192 bytes) seal, DES-EDE3-CBC-SHA1 (16384 bytes) seal, AES-128-CBC-SHA1 (16 bytes) seal, AES-128-CBC-SHA1 (256 bytes) seal, AES-128-CBC-SHA1 (1350 bytes) seal, AES-128-CBC-SHA1 (8192 bytes) seal, AES-128-CBC-SHA1 (16384 bytes) seal, AES-256-CBC-SHA1 (16 bytes) seal, AES-256-CBC-SHA1 (256 bytes) seal, AES-256-CBC-SHA1 (1350 bytes) seal, AES-256-CBC-SHA1 (8192 bytes) seal, AES-256-CBC-SHA1 (16384 bytes) seal, AES-128-CBC-SHA1 (16 bytes) open, AES-128-CBC-SHA1 (256 bytes) open, AES-128-CBC-SHA1 (1350 bytes) open, AES-128-CBC-SHA1 (8192 bytes) open, AES-128-CBC-SHA1 (16384 bytes) open, AES-256-CBC-SHA1 (16 bytes) open, AES-256-CBC-SHA1 (256 bytes) open, AES-256-CBC-SHA1 (1350 bytes) open, AES-256-CBC-SHA1 (8192 bytes) open, AES-256-CBC-SHA1 (16384 bytes) open, AES-128-GCM-SIV (16 bytes) seal, AES-128-GCM-SIV (256 bytes) seal, AES-128-GCM-SIV (1350 bytes) seal, AES-128-GCM-SIV (8192 bytes) seal, AES-128-GCM-SIV (16384 bytes) seal, AES-256-GCM-SIV (16 bytes) seal, AES-256-GCM-SIV (256 bytes) seal, AES-256-GCM-SIV (1350 bytes) seal, AES-256-GCM-SIV (8192 bytes) seal, AES-256-GCM-SIV (16384 bytes) seal, AES-128-GCM-SIV (16 bytes) open, AES-128-GCM-SIV (256 bytes) open, AES-128-GCM-SIV (1350 bytes) open, AES-128-GCM-SIV (8192 bytes) open, AES-128-GCM-SIV (16384 bytes) open, AES-256-GCM-SIV (16 bytes) open, AES-256-GCM-SIV (256 bytes) open, AES-256-GCM-SIV (1350 bytes) open, AES-256-GCM-SIV (8192 bytes) open, AES-256-GCM-SIV (16384 bytes) open, AES-128-CCM-Bluetooth (16 bytes) seal, AES-128-CCM-Bluetooth (256 bytes) seal, AES-128-CCM-Bluetooth (1350 bytes) seal, AES-128-CCM-Bluetooth (8192 bytes) seal, AES-128-CCM-Bluetooth (16384 bytes) seal, Ed25519 key generation, Ed25519 signing, Ed25519 verify, Curve25519 base-point multiplication, Curve25519 arbitrary point multiplication, SPAKE2 over Ed25519, HRSS generate, HRSS encap, HRSS decap, BLAKE2b-256 (16 bytes), BLAKE2b-256 (256 bytes), BLAKE2b-256 (1350 bytes), BLAKE2b-256 (8192 bytes), BLAKE2b-256 (16384 bytes), hash-to-curve P384_XMD:SHA-512_SSWU_RO_, hash-to-scalar P384_XMD:SHA-512, TrustToken-Exp1-Batch1 generate_key, TrustToken-Exp1-Batch1 begin_issuance, TrustToken-Exp1-Batch1 issue, TrustToken-Exp1-Batch1 finish_issuance, TrustToken-Exp1-Batch1 begin_redemption, TrustToken-Exp1-Batch1 redeem, TrustToken-Exp1-Batch1 finish_redemption, TrustToken-Exp1-Batch10 generate_key, TrustToken-Exp1-Batch10 begin_issuance, TrustToken-Exp1-Batch10 issue, TrustToken-Exp1-Batch10 finish_issuance, TrustToken-Exp1-Batch10 begin_redemption, TrustToken-Exp1-Batch10 redeem, TrustToken-Exp1-Batch10 finish_redemption, TrustToken-Exp2VOfPRF-Batch1 generate_key, TrustToken-Exp2VOfPRF-Batch1 begin_issuance, TrustToken-Exp2VOfPRF-Batch1 issue, TrustToken-Exp2VOfPRF-Batch1 finish_issuance, TrustToken-Exp2VOfPRF-Batch1 begin_redemption, TrustToken-Exp2VOfPRF-Batch1 redeem, TrustToken-Exp2VOfPRF-Batch1 finish_redemption, TrustToken-Exp2VOPRF-Batch10 generate_key, TrustToken-Exp2VOPRF-Batch10 begin_issuance, TrustToken-Exp2VOPRF-Batch10 issue, TrustToken-Exp2VOPRF-Batch10 finish_issuance, TrustToken-Exp2VOPRF-Batch10 begin_redemption, TrustToken-Exp2VOPRF-Batch10 redeem, TrustToken-Exp2VOPRF-Batch10 finish_redemption, TrustToken-Exp2PMB-Batch1 generate_key, TrustToken-Exp2PMB-Batch1 begin_issuance, TrustToken-Exp2PMB-Batch1 issue, TrustToken-Exp2PMB-Batch1 finish_issuance, TrustToken-Exp2PMB-Batch1 begin_redemption, TrustToken-Exp2PMB-Batch1 redeem, TrustToken-Exp2PMB-Batch1 finish_redemption, TrustToken-Exp2PMB-Batch10 generate_key, TrustToken-Exp2PMB-Batch10 begin_issuance, TrustToken-Exp2PMB-Batch10 issue, TrustToken-Exp2PMB-Batch10 finish_issuance, TrustToken-Exp2PMB-Batch10 begin_redemption, TrustToken-Exp2PMB-Batch10 redeem, TrustToken-Exp2PMB-Batch10 finish_redemption, base64 decode, SipHash-2-4 (16 bytes), SipHash-2-4 (256 bytes), SipHash-2-4 (1350 bytes), SipHash-2-4 (8192 bytes), SipHash-2-4 (16384 bytes)
```

Reply:
Does `significant_regressions` not always return true?

This line may always return a mixed of True and False array given this doc https://numpy.org/doc/stable/reference/generated/numpy.isin.html#
```py
evaluation = np.isin(descriptions, ignored_fields)
```

Reply:
Discussed offline.

`compared_df` only contains entries for dimensions that saw a >15% regression